spec/Section 3 -- Type System.md

```diff
@@ -263,74 +263,108 @@ TypeDefinition :
   - InputObjectTypeDefinition
 
 The fundamental unit of any GraphQL Schema is the type. There are six kinds
-of named type definitions in GraphQL, and two wrapping types.
+of *named type* and two kinds of *wrapped type* definitions in GraphQL.
 
-The most basic type is a `Scalar`. A scalar represents a primitive value, like
-a string or an integer. Oftentimes, the possible responses for a scalar field
-are enumerable. GraphQL offers an `Enum` type in those cases, where the type
-specifies the space of valid responses.
+:: A *named type* is one of the six types defined with a name: *Scalar type*,
+*Enum type*, *Object type*, *Interface type*, *Union type*,
+and *Input Object type*.
 
-Scalars and Enums form the leaves in response trees; the intermediate levels are
-`Object` types, which define a set of fields, where each field is another
-type in the system, allowing the definition of arbitrary type hierarchies.
 
-GraphQL supports two abstract types: interfaces and unions.
+### Leaf and Composite Types
 
-An `Interface` defines a list of fields; `Object` types and other Interface
-types which implement this Interface are guaranteed to implement those fields.
-Whenever a field claims it will return an Interface type, it will return a
-valid implementing Object type during execution.
+:: A *leaf type* is found at the leaves in a response tree. There are two kinds
+of leaf types: *Scalar type* and *Enum type*.
 
-A `Union` defines a list of possible types; similar to interfaces, whenever the
-type system claims a union will be returned, one of the possible types will be
-returned.
+The most basic *leaf type* is the *Scalar type*. A scalar represents a primitive
+value, like a string or an integer. GraphQL defines a set of *built-in scalars*
+and also allows a service to additionally provide any *custom scalar*.
 
-Finally, oftentimes it is useful to provide complex structs as inputs to
-GraphQL field arguments or variables; the `Input Object` type allows the schema
-to define exactly what data is expected.
+Oftentimes, the possible values for a *leaf type* are enumerable. GraphQL offers
+the *Enum type* in those cases, where the each enum type definition specifies
+the set of possible values.
 
+:: A *composite type* compose together leaf types and other composite types as a
```

Comment:
A leaf type is scalar/enum, not including wrapped. Same with composite types being the set described here, not including wrapped.

This is good feedback on this paragraph being falsely precise - since we can compose leaf or composite types (leaf + composite = named) and any wrapped of either of those (named + wrapped(named) = all types). Thats just all types.

Going to replace with something simple:

> A *composite type* composes other types together as a set of named fields

Reply:
Sounds good to me 👍